internal/oci/runtime_oci.go

```diff
@@ -112,13 +113,22 @@ func (r *runtimeOCI) CreateContainer(ctx context.Context, c *Container, cgroupPa
 		"-P", c.conmonPidFilePath(),
 		"-p", filepath.Join(c.bundlePath, "pidfile"),
 		"--persist-dir", c.dir,
-		"-r", r.handler.RuntimePath,
 		"--runtime-arg", fmt.Sprintf("%s=%s", rootFlag, r.root),
 		"--socket-dir-path", r.config.ContainerAttachSocketDir,
 		"--syslog",
 		"-u", c.ID(),
 	}
 
+	imageResult, err := getImageDetailsForContainer(r.Runtime, ctx, c)
```

Comment:
This might be somewhat expensive, especially given the fact it is needed for wasm only, but the code is executed every time.

Reply:
Yeah, I agree with you. At the same time, I feel getting platform details of an image required creating an `ImageServer` that needs a bunch of other stuff to compute. I'm happy to hear the better suggestions here.

Reply:
instead, I think you should execute this in server/ where we already have the image store initialized. There, you can maybe call a oci.Container function SetArch(), and here we can just call c.Arch() and see if it's defined in the runtime table